mypy/semanal.py

```diff
@@ -720,6 +720,35 @@ def analyze_class_body(self, defn: ClassDef) -> Iterator[bool]:
                 self.calculate_abstract_status(defn.info)
                 self.setup_type_promotion(defn)
 
+                for decorator in defn.decorators:
+                    fullname = None
+                    if isinstance(decorator, CallExpr):
+                        if isinstance(decorator.callee, RefExpr):
+                            fullname = decorator.callee.fullname
+                    elif isinstance(decorator, NameExpr):
+                        fullname = decorator.fullname
+
+                    if fullname:
+                        hook = self.plugin.get_class_decorator_hook(fullname)
+                        if hook:
+                            hook(ClassDefContext(defn, self))
+
+                if defn.metaclass:
+                    metaclass_name = None
+                    if isinstance(defn.metaclass, NameExpr):
+                        metaclass_name = defn.metaclass.name
+                    elif isinstance(defn.metaclass, MemberExpr):
+                        metaclass_name = get_member_expr_fullname(
+                            defn.metaclass)
+                    hook = self.plugin.get_class_metaclass_hook(metaclass_name)
+                    if hook:
+                        hook(ClassDefContext(defn, self))
+
+                for type_info in defn.info.bases:
+                    hook = self.plugin.get_class_base_hook(type_info.type.fullname())
+                    if hook:
+                        hook(ClassDefContext(defn, self))
+
```

Comment:
I would factor out this whole block into a separate method, and call it here like ``self.apply_plugin_hooks(defn)`` or similar.

Reply:
Sounds good. Will do.